### PR TITLE
ddl, session: add tidb_enable_exchange_partition variable (#22638)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #22638 to release-5.0-rc
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/22706
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/22706:release-5.0-rc-9e3c0649e43d
```

---

### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?

Add tidb_enable_exchange_partition variable to enable or disable exchange partition.
Disabled by default.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- add tidb_enable_exchange_partition variable

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -6489,3 +6489,68 @@ func (s *testDBSuite4) TestUnsupportedAlterTableOption(c *C) {
 	tk.MustExec("create table t(a char(10) not null,b char(20)) shard_row_id_bits=6;")
 	tk.MustGetErrCode("alter table t pre_split_regions=6;", errno.ErrUnsupportedDDLOperation)
 }
+<<<<<<< HEAD
```
Comment (qw4990) [inline ddl/db_test.go:6492]: @rebelice Please resolve conflicts

**Review by qw4990 — COMMENTED** [review]
LGTM

**Review by zz-jason — APPROVED** [review]
LGTM

**Pull request merged**